template.tpl

```diff
@@ -333,14 +370,18 @@ var eventMetadata = {
 };
 
 // Certain events don't support certain params, so we conditionally set them
-if (data.eventType != "AddToCart" && data.eventType != "AddToWishlist") {
+if (data.eventType != "AddToCart" && data.eventType != "AddToWishlist" && data.transactionId) {
   eventMetadata.transactionId = data.transactionId;
 }
 
-if (data.eventType != "SignUp" && data.eventType != "Lead") {
+if (data.eventType != "SignUp" && data.eventType != "Lead" && data.itemCount) {
```

Comment:
If `itemCount` is zero, then the value won't get set.

Where an empty string for `transactionId` or for `customEventName` gives the correct behavior -- those should not be set if they're blank -- a zero-value item count is meaningful.

Does teh Google allow the template to distinguish between blank or unset and a zero value?  For example, does it pass `null` or is the attribute `undefined`?

Reply:
That is, JavaScript uses [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) and [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) values in logical expressions.

Reply:
Ah that's a good point. Just tested in GTM and it looks like there's no difference. In that case I'm going to leave the previous logic as is, since we handle the values downstream.